modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/field/predicate/provider/MultiselectPicklistFieldPredicateProvider.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.rest.internal.odata.filter.expression.field.predicate.provider;
+
+import com.liferay.object.constants.ObjectFieldConstants;
+import com.liferay.object.odata.filter.expression.field.predicate.provider.FieldPredicateProvider;
+import com.liferay.petra.sql.dsl.Column;
+import com.liferay.petra.sql.dsl.DSLFunctionFactoryUtil;
+import com.liferay.petra.sql.dsl.expression.Expression;
+import com.liferay.petra.sql.dsl.expression.Predicate;
+import com.liferay.petra.sql.dsl.spi.expression.Scalar;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.odata.filter.expression.BinaryExpression;
+import com.liferay.portal.odata.filter.expression.MethodExpression;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Sergio Jiménez del Coso
+ */
+@Component(
+	property = "field.predicate.provider.key=" + ObjectFieldConstants.BUSINESS_TYPE_MULTISELECT_PICKLIST,
+	service = FieldPredicateProvider.class
+)
+public class MultiselectPicklistFieldPredicateProvider
+	implements FieldPredicateProvider {
+
+	@Override
+	public Predicate getBinaryExpressionPredicate(
+		Function<String, Column<?, ?>> objectDefinitionColumnSupplier,
+		Object left, long objectDefinitionId,
+		BinaryExpression.Operation operation, Object right) {
+
+		Expression<String> columnFieldExpression =
+			(Expression<String>)objectDefinitionColumnSupplier.apply(
+				String.valueOf(left));
+
+		if (Objects.equals(operation, BinaryExpression.Operation.EQ)) {
+			Expression<String> formatedColumnExpression =
+				_getFormatedColumnExpression(columnFieldExpression);
+
+			return formatedColumnExpression.like(
+				_getFieldValueExpression(right, null));
+		}
+		else if (Objects.equals(operation, BinaryExpression.Operation.NE)) {
+			return columnFieldExpression.neq(String.valueOf(right));
```

Comment:
After taking a view at some examples like `keywords` and `taxonomyCategoryIds` properties, I implemented a similar functionality, if the user filter with NotEquals operation given a value (i.e `2`), it will return all the items that has not only the `2,` but it can return some items which contain the value of `2`. However, it's true that if the user hardcodes the filter like: `?filter=multiField/any(k:k ne '2, 12')`, this could throw an INTERNAL SERVER ERROR because it can not create the filter due to the commas. That's the reason why I implemented the NotEquals operation, does it make sense to support it @ccorreagg ?